prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -58,10 +78,181 @@ class Hl7Serializer(val metadata: Metadata) {
         outputStream.write(createFooters(report).toByteArray())
     }
 
+    /*
+     * Read in a file
+     */
+    fun convertBatchMessagesToMap(message: String, schema: Schema): Hl7Mapping {
+        val mappedRows: MutableMap<String, MutableList<String>> = mutableMapOf()
+        val errors = mutableListOf<String>()
+        val warnings = mutableListOf<String>()
+        val rowResults = mutableListOf<RowResult>()
+        val reg = "(\r|\n)".toRegex()
+        val cleanedMessage = reg.replace(message, "\r")
+        val messageLines = cleanedMessage.split("\r")
```

Comment:
In order to properly process a message, I needed to create a regex that replaces `\r` and `\n` with just `\r` and then split on that. If I don't do this, HAPI just reads the entire message as non-standard MSH additions.